shared/metrics/src/main/java/io/pravega/shared/metrics/OpStatsLoggerImpl.java

```diff
@@ -90,15 +92,22 @@ public void clear() {
 
     @Override
     public OpStatsData toOpStatsData() {
-        long numFailed = fail.getCount();
-        long numSuccess = success.getCount();
-        Snapshot s = success.getSnapshot();
-        double avgLatencyMillis = s.getMean();
+        long numFailed = fail.count();
+        long numSuccess = success.count();
+        HistogramSnapshot snapshot = success.takeSnapshot();
+        double avgLatencyMillis = success.mean(TimeUnit.SECONDS);
 
         EnumMap<OpStatsData.Percentile, Long> percentileLongMap  =
-                new EnumMap<OpStatsData.Percentile, Long>(OpStatsData.Percentile.class);
+                new EnumMap<>(OpStatsData.Percentile.class);
+        ValueAtPercentile[] values = snapshot.percentileValues();
+
+        //TODO: refactor OpStatsData.Percentile based on micrometer implementation
         for (OpStatsData.Percentile percent : OpStatsData.PERCENTILESET) {
-            percentileLongMap.put(percent, (long) s.getValue(percent.getValue() / 100));
+            for (ValueAtPercentile vp : snapshot.percentileValues()) {
+                if (vp.percentile() == percent.getValue() / 100) {
```

Comment:
store `percent.getValue()/100` into a variable and use it here. Otherwise you'll end up calculating this over and over again in every loop iteration.

Reply:
Yes.
I'm changing OpStatsData.PercentileSet to be based on Micrometer's set, the code will be optimized along the way.